java.io.EOFException: HttpConnectionOverHTTP@6f56146f::DecryptedEndPoint@71b6e0a0... error

If an HTTP client instance is used in a multithreading environment (e.g. multiple requests are sent via the same object in parallel) we get occasionally the following error:

> java.io.EOFException: HttpConnectionOverHTTP@43f11127::DecryptedEndPoint@58c1c452{localhost/127.0.0.1:8080<->/127.0.0.1:50734,OPEN,fill=-,flush=P,to=1/0}


Code to reproduce can be found here: https://github.com/Andremoniy/jetty.httpclient.sslissue

The full stacktrace:

```
java.util.concurrent.ExecutionException: java.io.EOFException: HttpConnectionOverHTTP@43f11127::DecryptedEndPoint@58c1c452{localhost/127.0.0.1:8080<->/127.0.0.1:50734,OPEN,fill=-,flush=P,to=1/0}
	at org.eclipse.jetty.client.util.FutureResponseListener.getResult(FutureResponseListener.java:118)
	at org.eclipse.jetty.client.util.FutureResponseListener.get(FutureResponseListener.java:101)
	at org.eclipse.jetty.client.HttpRequest.send(HttpRequest.java:683)
	at com.github.andremoniy.jetty.httpclient.sslissue.Bootstrap.lambda$main$1(Bootstrap.java:60)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: 
java.io.EOFException: HttpConnectionOverHTTP@43f11127::DecryptedEndPoint@58c1c452{localhost/127.0.0.1:8080<->/127.0.0.1:50734,OPEN,fill=-,flush=P,to=1/0}
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.earlyEOF(HttpReceiverOverHTTP.java:338)
	at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:1552)
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.shutdown(HttpReceiverOverHTTP.java:209)
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.process(HttpReceiverOverHTTP.java:147)
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.receive(HttpReceiverOverHTTP.java:73)
	at org.eclipse.jetty.client.http.HttpChannelOverHTTP.receive(HttpChannelOverHTTP.java:133)
	at org.eclipse.jetty.client.http.HttpConnectionOverHTTP.onFillable(HttpConnectionOverHTTP.java:155)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at org.eclipse.jetty.io.ssl.SslConnection$DecryptedEndPoint$IncompleteWriteCallback.succeeded(SslConnection.java:1309)
	at org.eclipse.jetty.io.WriteFlusher.write(WriteFlusher.java:293)
	at org.eclipse.jetty.io.AbstractEndPoint.write(AbstractEndPoint.java:380)
	at org.eclipse.jetty.io.ssl.SslConnection$DecryptedEndPoint.onIncompleteFlush(SslConnection.java:1068)
	at org.eclipse.jetty.io.AbstractEndPoint$2.onIncompleteFlush(AbstractEndPoint.java:54)
	at org.eclipse.jetty.io.WriteFlusher.write(WriteFlusher.java:285)
	at org.eclipse.jetty.io.AbstractEndPoint.write(AbstractEndPoint.java:380)
	at org.eclipse.jetty.client.http.HttpSenderOverHTTP$HeadersCallback.process(HttpSenderOverHTTP.java:268)
	at org.eclipse.jetty.util.IteratingCallback.processing(IteratingCallback.java:241)
	at org.eclipse.jetty.util.IteratingCallback.iterate(IteratingCallback.java:224)
	at org.eclipse.jetty.client.http.HttpSenderOverHTTP.sendHeaders(HttpSenderOverHTTP.java:62)
	at org.eclipse.jetty.client.HttpSender.send(HttpSender.java:214)
	at org.eclipse.jetty.client.http.HttpChannelOverHTTP.send(HttpChannelOverHTTP.java:85)
	at org.eclipse.jetty.client.HttpChannel.send(HttpChannel.java:128)
	at org.eclipse.jetty.client.HttpConnection.send(HttpConnection.java:201)
	at org.eclipse.jetty.client.http.HttpConnectionOverHTTP$Delegate.send(HttpConnectionOverHTTP.java:255)
	at org.eclipse.jetty.client.http.HttpConnectionOverHTTP.send(HttpConnectionOverHTTP.java:122)
	at org.eclipse.jetty.client.http.HttpDestinationOverHTTP.send(HttpDestinationOverHTTP.java:38)
	at org.eclipse.jetty.client.HttpDestination.process(HttpDestination.java:346)
	at org.eclipse.jetty.client.HttpDestination.process(HttpDestination.java:304)
	at org.eclipse.jetty.client.HttpDestination.send(HttpDestination.java:294)
	at org.eclipse.jetty.client.HttpDestination.send(HttpDestination.java:269)
	at org.eclipse.jetty.client.HttpDestination.send(HttpDestination.java:246)
	at org.eclipse.jetty.client.HttpClient.send(HttpClient.java:576)
	at org.eclipse.jetty.client.HttpRequest.send(HttpRequest.java:726)
	at org.eclipse.jetty.client.HttpRequest.send(HttpRequest.java:679)
	at com.github.andremoniy.jetty.httpclient.sslissue.Bootstrap.lambda$main$1(Bootstrap.java:60)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

Thanks for your reply, @joakime! 
I've just tried to switch off TLS 1.3 by adding
`        sslContextFactory.setExcludeProtocols("TLSv1.3");` (please see commit in my repo: https://github.com/Andremoniy/jetty.httpclient.sslissue/commit/cb3c8ea27bdfa72fcc49ff6b62dae7ea187db85a)

and then I've made two runs and got one new error (`java.io.IOException: Broken pipe`) and the very same error as it was before on the 2nd run. Please see below:
The first run:
```
2019-07-19 12:04:02.841:INFO::main: Logging initialized @1089ms to org.eclipse.jetty.util.log.StdErrLog
2019-07-19 12:04:03.471:INFO:oejs.Server:main: jetty-9.4.19.v20190610; built: 2019-06-10T16:30:51.723Z; git: afcf563148970e98786327af5e07c261fda175d3; jvm 11.0.4+11
2019-07-19 12:04:03.538:INFO:oejs.session:main: DefaultSessionIdManager workerName=node0
2019-07-19 12:04:03.538:INFO:oejs.session:main: No SessionScavenger set, using defaults
2019-07-19 12:04:03.543:INFO:oejs.session:main: node0 Scavenging every 600000ms
2019-07-19 12:04:04.741:INFO:oejsh.ContextHandler:main: Started o.e.j.s.ServletContextHandler@73d6d0c{/,null,AVAILABLE}
2019-07-19 12:04:04.767:INFO:oejus.SslContextFactory:main: x509=X509@607b2792(sitename,h=[],w=[]) for Server@7f9e1534[provider=null,keyStore=null,trustStore=null]
2019-07-19 12:04:05.024:INFO:oejs.AbstractConnector:main: Started ServerConnector@7ce026d3{SSL,[ssl, http/1.1]}{0.0.0.0:8080}
2019-07-19 12:04:05.025:INFO:oejs.Server:main: Started @3273ms
2019-07-19 12:04:05.065:WARN:oejusS.config:main: Trusting all certificates configured for Client@1869f114[provider=null,keyStore=null,trustStore=null]
2019-07-19 12:04:05.065:WARN:oejusS.config:main: No Client EndPointIdentificationAlgorithm configured for Client@1869f114[provider=null,keyStore=null,trustStore=null]
2019-07-19 12:04:05.072:INFO:cgajhs.Bootstrap:main: Running 8 threads
2019-07-19 12:08:28.063:WARN:cgajhs.Bootstrap:Thread-20: java.io.IOException: Broken pipe
java.util.concurrent.ExecutionException: java.io.IOException: Broken pipe
	at org.eclipse.jetty.client.util.FutureResponseListener.getResult(FutureResponseListener.java:118)
	at org.eclipse.jetty.client.util.FutureResponseListener.get(FutureResponseListener.java:101)
	at org.eclipse.jetty.client.HttpRequest.send(HttpRequest.java:683)
	at org.eclipse.jetty.client.HttpClient.GET(HttpClient.java:362)
	at org.eclipse.jetty.client.HttpClient.GET(HttpClient.java:347)
	at com.github.andremoniy.jetty.httpclient.sslissue.Bootstrap.lambda$main$0(Bootstrap.java:49)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: 
java.io.IOException: Broken pipe
	at org.eclipse.jetty.io.ssl.SslConnection$DecryptedEndPoint.flush(SslConnection.java:927)
	at org.eclipse.jetty.io.WriteFlusher.flush(WriteFlusher.java:393)
	at org.eclipse.jetty.io.WriteFlusher.completeWrite(WriteFlusher.java:349)
	at org.eclipse.jetty.io.ssl.SslConnection$DecryptedEndPoint.lambda$fill$1(SslConnection.java:670)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:781)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:917)
	at java.base/java.lang.Thread.run(Thread.java:834)
Exception in thread "Thread-20" 2019-07-19 12:08:28.067:WARN:cgajhs.Bootstrap:Thread-19: 
java.lang.InterruptedException
	at java.base/java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1040)
	at java.base/java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1345)
	at java.base/java.util.concurrent.CountDownLatch.await(CountDownLatch.java:232)
	at org.eclipse.jetty.client.util.FutureResponseListener.get(FutureResponseListener.java:100)
	at org.eclipse.jetty.client.HttpRequest.send(HttpRequest.java:683)
	at com.github.andremoniy.jetty.httpclient.sslissue.Bootstrap.lambda$main$1(Bootstrap.java:60)
	at java.base/java.lang.Thread.run(Thread.java:834)
java.lang.IllegalStateException: java.util.concurrent.ExecutionException: java.io.IOException: Broken pipe
	at com.github.andremoniy.jetty.httpclient.sslissue.Bootstrap.lambda$main$0(Bootstrap.java:53)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.util.concurrent.ExecutionException: java.io.IOException: Broken pipe
	at org.eclipse.jetty.client.util.FutureResponseListener.getResult(FutureResponseListener.java:118)
	at org.eclipse.jetty.client.util.FutureResponseListener.get(FutureResponseListener.java:101)
	at org.eclipse.jetty.client.HttpRequest.send(HttpRequest.java:683)
	at org.eclipse.jetty.client.HttpClient.GET(HttpClient.java:362)
	at org.eclipse.jetty.client.HttpClient.GET(HttpClient.java:347)
	at com.github.andremoniy.jetty.httpclient.sslissue.Bootstrap.lambda$main$0(Bootstrap.java:49)
	... 1 more
Caused by: java.io.IOException: Broken pipe
	at org.eclipse.jetty.io.ssl.SslConnection$DecryptedEndPoint.flush(SslConnection.java:927)
	at org.eclipse.jetty.io.WriteFlusher.flush(WriteFlusher.java:393)
	at org.eclipse.jetty.io.WriteFlusher.completeWrite(WriteFlusher.java:349)
	at org.eclipse.jetty.io.ssl.SslConnection$DecryptedEndPoint.lambda$fill$1(SslConnection.java:670)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:781)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:917)
	... 1 more
```

and the 2nd run:

```
2019-07-19 12:09:40.718:INFO::main: Logging initialized @999ms to org.eclipse.jetty.util.log.StdErrLog
2019-07-19 12:09:41.142:INFO:oejs.Server:main: jetty-9.4.19.v20190610; built: 2019-06-10T16:30:51.723Z; git: afcf563148970e98786327af5e07c261fda175d3; jvm 11.0.4+11
2019-07-19 12:09:41.195:INFO:oejs.session:main: DefaultSessionIdManager workerName=node0
2019-07-19 12:09:41.195:INFO:oejs.session:main: No SessionScavenger set, using defaults
2019-07-19 12:09:41.199:INFO:oejs.session:main: node0 Scavenging every 600000ms
2019-07-19 12:09:42.184:INFO:oejsh.ContextHandler:main: Started o.e.j.s.ServletContextHandler@73d6d0c{/,null,AVAILABLE}
2019-07-19 12:09:42.213:INFO:oejus.SslContextFactory:main: x509=X509@607b2792(sitename,h=[],w=[]) for Server@7f9e1534[provider=null,keyStore=null,trustStore=null]
2019-07-19 12:09:42.553:INFO:oejs.AbstractConnector:main: Started ServerConnector@7ce026d3{SSL,[ssl, http/1.1]}{0.0.0.0:8080}
2019-07-19 12:09:42.553:INFO:oejs.Server:main: Started @2835ms
2019-07-19 12:09:42.598:WARN:oejusS.config:main: Trusting all certificates configured for Client@1869f114[provider=null,keyStore=null,trustStore=null]
2019-07-19 12:09:42.598:WARN:oejusS.config:main: No Client EndPointIdentificationAlgorithm configured for Client@1869f114[provider=null,keyStore=null,trustStore=null]
2019-07-19 12:09:42.610:INFO:cgajhs.Bootstrap:main: Running 8 threads
2019-07-19 12:13:48.047:WARN:cgajhs.Bootstrap:Thread-23: java.io.EOFException: HttpConnectionOverHTTP@4e9355be::DecryptedEndPoint@2bb44ed6{localhost/127.0.0.1:8080<->/127.0.0.1:50816,OPEN,fill=-,flush=C,to=46835/0}
java.util.concurrent.ExecutionException: java.io.EOFException: HttpConnectionOverHTTP@4e9355be::DecryptedEndPoint@2bb44ed6{localhost/127.0.0.1:8080<->/127.0.0.1:50816,OPEN,fill=-,flush=C,to=46835/0}
	at org.eclipse.jetty.client.util.FutureResponseListener.getResult(FutureResponseListener.java:118)
	at org.eclipse.jetty.client.util.FutureResponseListener.get(FutureResponseListener.java:101)
	at org.eclipse.jetty.client.HttpRequest.send(HttpRequest.java:683)
	at com.github.andremoniy.jetty.httpclient.sslissue.Bootstrap.lambda$main$1(Bootstrap.java:60)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: 
java.io.EOFException: HttpConnectionOverHTTP@4e9355be::DecryptedEndPoint@2bb44ed6{localhost/127.0.0.1:8080<->/127.0.0.1:50816,OPEN,fill=-,flush=C,to=46835/0}
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.earlyEOF(HttpReceiverOverHTTP.java:338)
	at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:1552)
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.shutdown(HttpReceiverOverHTTP.java:209)
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.process(HttpReceiverOverHTTP.java:147)
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.receive(HttpReceiverOverHTTP.java:73)
	at org.eclipse.jetty.client.http.HttpChannelOverHTTP.receive(HttpChannelOverHTTP.java:133)
	at org.eclipse.jetty.client.http.HttpConnectionOverHTTP.onFillable(HttpConnectionOverHTTP.java:155)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at org.eclipse.jetty.io.ssl.SslConnection$DecryptedEndPoint.onFillable(SslConnection.java:427)
	at org.eclipse.jetty.io.ssl.SslConnection.onFillable(SslConnection.java:321)
	at org.eclipse.jetty.io.ssl.SslConnection$2.succeeded(SslConnection.java:159)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:781)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:917)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

And also you might be interested in the run performed with TLSv1.3 enabled. From wireshark perspective they look different, though TLSv1.2 seems to be used in the communication. 
Please find them attached. 
[51100_wireshark.pcapng.zip](https://github.com/eclipse/jetty.project/files/3430274/51100_wireshark.pcapng.zip)
[51100.log.zip](https://github.com/eclipse/jetty.project/files/3430275/51100.log.zip)
```
2019-07-25 08:55:15.524:DBUG:oejh.HttpGenerator:qtp225344427-63: Date: Thu, 25 Jul 2019 06:55:15 GMT<|Content-Length: 2<|Content-Type: text/html<|<|
java.lang.IllegalStateException: java.util.concurrent.ExecutionException: java.io.EOFException: HttpConnectionOverHTTP@331f2f10::DecryptedEndPoint@13ae8156{localhost/127.0.0.1:50734<->/127.0.0.1:51100,OPEN,fill=-,flush=-,to=414346/0}
	at com.github.andremoniy.jetty.httpclient.sslissue.BootstrapWithJettyServer.lambda$main$1(BootstrapWithJettyServer.java:66)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.util.concurrent.ExecutionException: java.io.EOFException: HttpConnectionOverHTTP@331f2f10::DecryptedEndPoint@13ae8156{localhost/127.0.0.1:50734<->/127.0.0.1:51100,OPEN,fill=-,flush=-,to=414346/0}
	at org.eclipse.jetty.client.util.FutureResponseListener.getResult(FutureResponseListener.java:118)
	at org.eclipse.jetty.client.util.FutureResponseListener.get(FutureResponseListener.java:101)
	at org.eclipse.jetty.client.HttpRequest.send(HttpRequest.java:683)
	at com.github.andremoniy.jetty.httpclient.sslissue.BootstrapWithJettyServer.lambda$main$1(BootstrapWithJettyServer.java:62)
	... 1 more
Caused by: java.io.EOFException: HttpConnectionOverHTTP@331f2f10::DecryptedEndPoint@13ae8156{localhost/127.0.0.1:50734<->/127.0.0.1:51100,OPEN,fill=-,flush=-,to=414346/0}
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.earlyEOF(HttpReceiverOverHTTP.java:338)
	at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:1552)
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.shutdown(HttpReceiverOverHTTP.java:209)
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.process(HttpReceiverOverHTTP.java:147)
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.receive(HttpReceiverOverHTTP.java:73)
	at org.eclipse.jetty.client.http.HttpChannelOverHTTP.receive(HttpChannelOverHTTP.java:133)
	at org.eclipse.jetty.client.http.HttpConnectionOverHTTP.onFillable(HttpConnectionOverHTTP.java:155)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at org.eclipse.jetty.io.ssl.SslConnection$DecryptedEndPoint.onFillable(SslConnection.java:427)
	at org.eclipse.jetty.io.ssl.SslConnection.onFillable(SslConnection.java:321)
	at org.eclipse.jetty.io.ssl.SslConnection$2.succeeded(SslConnection.java:159)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:781)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:917)
	... 1 more
```

@joakime Thanks! Tried it, but it does not seem to affect the problem.

This is the typical stacktrace:
```
Caused by: java.io.EOFException: HttpConnectionOverHTTP@6d67c334::DecryptedEndPoint@da8b43f{/172.30.0.4:8900<->/172.30.0.1:56590,OPEN,fill=-,flush=W,to=6/30000}
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.earlyEOF(HttpReceiverOverHTTP.java:338)
	at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:1552)
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.shutdown(HttpReceiverOverHTTP.java:209)
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.process(HttpReceiverOverHTTP.java:147)
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.access$200(HttpReceiverOverHTTP.java:40)
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP$1.resume(HttpReceiverOverHTTP.java:273)
	at org.eclipse.jetty.util.CompletableCallback.succeeded(CompletableCallback.java:82)
	at org.eclipse.jetty.util.Callback$Nested.succeeded(Callback.java:261)
	at org.eclipse.jetty.util.CountingCallback.succeeded(CountingCallback.java:68)
	at org.eclipse.jetty.client.util.InputStreamResponseListener$Input.read(InputStreamResponseListener.java:331)
	at java.io.InputStream.skip(InputStream.java:224)
```

@sbordet Here is the stack trace - 
```Caused by: java.util.concurrent.ExecutionException: java.io.EOFException: HttpConnectionOverHTTP@43e5a4db::DecryptedEndPoint@6e415863{charon-downstream-fnt.i.test.fctsordering.expedia.com/10.34.25.208:443<->/172.17.0.4:56240,CLOSED,fill=-,flush=C,to=3/20000}\n\tat org.eclipse.jetty.client.util.FutureResponseListener.getResult(FutureResponseListener.java:118)\n\tat org.eclipse.jetty.client.util.FutureResponseListener.get(FutureResponseListener.java:101)\n\tat org.eclipse.jetty.client.HttpRequest.send(HttpRequest.java:683)\n\tat com.expedia.om.scube.impl.ProxyHttpInvoker.invoke(ProxyHttpInvoker.java:107)\n\t... 67 more\nCaused by: java.io.EOFException: HttpConnectionOverHTTP@43e5a4db::DecryptedEndPoint@6e415863{charon-downstream-fnt.i.test.fctsordering.expedia.com/10.34.25.208:443<->/172.17.0.4:56240,CLOSED,fill=-,flush=C,to=3/20000}\n\tat org.eclipse.jetty.client.http.HttpReceiverOverHTTP.earlyEOF(HttpReceiverOverHTTP.java:338)\n\tat org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:1551)\n\tat org.eclipse.jetty.client.http.HttpReceiverOverHTTP.shutdown(HttpReceiverOverHTTP.java:209)\n\tat org.eclipse.jetty.client.http.HttpReceiverOverHTTP.process(HttpReceiverOverHTTP.java:147)\n\tat org.eclipse.jetty.client.http.HttpReceiverOverHTTP.receive(HttpReceiverOverHTTP.java:73)\n\tat org.eclipse.jetty.client.http.HttpChannelOverHTTP.receive(HttpChannelOverHTTP.java:133)\n\tat org.eclipse.jetty.client.http.HttpConnectionOverHTTP.onFillable(HttpConnectionOverHTTP.java:155)\n\tat org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305)\n\tat org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)\n\tat org.eclipse.jetty.io.ssl.SslConnection$1.run(SslConnection.java:144)\n\t... 3 more\n"```

--- a/.gitignore
+++ b/.gitignore
@@ -25,3 +25,6 @@ hs_err_pid*
 berbix-java/build
 demo-app/build
 .idea/
+.gradle/
+berbix-java/.gradle/
+berbix-java/.DS_Store

--- a/berbix-java/build.gradle
+++ b/berbix-java/build.gradle
@@ -2,9 +2,10 @@ plugins {
     id 'java'
     id 'maven-publish'
     id 'signing'
+    id 'java-library-distribution'
 }
 
-version '1.0.2'
+version '1.0.3'
 group 'com.berbix'
 
 repositories {
@@ -21,7 +22,7 @@ dependencies {
     implementation 'com.fasterxml.jackson.core:jackson-databind:2.12.2'
     implementation 'com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.12.2'
     implementation 'javax.xml.bind:jaxb-api:2.3.1'
-    implementation 'org.eclipse.jetty:jetty-client:9.4.38.v20210224'
+    implementation 'org.eclipse.jetty:jetty-client:9.4.49.v20220914'
     testImplementation 'org.junit.jupiter:junit-jupiter-api:5.6.0'
     testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine'
 }
@@ -38,6 +39,7 @@ java {
 publishing {
     publications {
         berbixJava(MavenPublication) {
+            artifact distZip
             from components.java
 
             pom {

